app/gateway/web/account/get_all_accounts_test.go

```diff
@@ -0,0 +1,86 @@
+package account
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	usecase "stoneBanking/app/application/usecase/account"
+	"stoneBanking/app/domain/entities/account"
+	"stoneBanking/app/domain/entities/token"
+	customError "stoneBanking/app/domain/errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_GetAll(t *testing.T) {
+	testCases := []struct {
+		name            string
+		accountUsecase  usecase.Usecase
+		tokenRepository token.Repository
+		wantCode        int
+		wantBody        []map[string]interface{}
+		wantErr         error
+	}{
+		{
+			name: "with at last one account existing, data from account is returned sucessfully",
+			accountUsecase: usecase.New(
+				&account.RepositoryMock{
+					GetAllFunc: func(ctx context.Context) ([]account.Account, error) {
+						return []account.Account{{
+							ID:         1,
+							Name:       "Joao do Rio",
+							ExternalID: "94b9c27e-2880-42e3-8988-62dceb6b6463",
+							CPF:        "761.647.810-78",
+							Secret:     "J0@0doR10",
+							Balance:    0,
+						}}, nil
+					},
+				},
+				&token.RepositoryMock{}),
+			tokenRepository: &token.RepositoryMock{},
+			wantCode:        200,
+			wantBody: []map[string]interface{}{{
+				"id":         "94b9c27e-2880-42e3-8988-62dceb6b6463",
+				"name":       "Joao do Rio",
+				"cpf":        "761.647.810-78",
+				"balance":    0,
+				"created_at": "0001-01-01T00:00:00Z",
+			}},
+			wantErr: nil,
+		},
+		{
+			name: "with one error when listing the accounts, return error for client",
+			accountUsecase: usecase.New(
+				&account.RepositoryMock{
+					GetAllFunc: func(ctx context.Context) ([]account.Account, error) {
+						return []account.Account{{}}, customError.ErrorAccountsListing
+					},
+				},
+				&token.RepositoryMock{}),
+			tokenRepository: &token.RepositoryMock{},
+			wantCode:        500,
+			wantErr:         customError.ErrorAccountsListing,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+
+			rec := httptest.NewRecorder()
+			req := httptest.NewRequest(http.MethodGet, "/account", nil)
+			controller := New(test.accountUsecase, test.tokenRepository)
+			controller.GetAll(rec, req)
+			assert.Equal(t, test.wantCode, rec.Code)
+
+			if test.wantBody != nil {
+				wantBodyJson, _ := json.Marshal(test.wantBody)
+				assert.JSONEq(t, string(wantBodyJson), rec.Body.String())
+			}
+
+			if test.wantErr != nil {
+				assert.Equal(t, (test.wantErr.Error() + "\n"), rec.Body.String())
```

Comment:
Não sei se é o caso, mas me parece que a sua API está retornado erros em formato que não é json, apenas uma mensagem no corpo da resposta.

Se for o caso, talvez fosse uma boa sempre retornar o erro em formato json também

Reply:
Será resolvido na Issue #65